src/v2/segment.v2.ejs

```diff
@@ -0,0 +1,159 @@
+async function fetchDestinations(writeKey) {
+  let destinations = await fetchDestinationForWriteKey(writeKey);
+
+  destinations = [
+    ...destinations
+      .reduce((map, item) => {
+        if (item.id === "Repeater") return map; // remove Segment Repeater destinations
+        map.has(item["id"]) || map.set(item["id"], item);
+        return map;
+      }, new Map()) // return object
+      .values()
+  ];
+
+  return destinations;
+}
+
+function getConsentAndLoadSegment() {
+  let oneTrustConsent = '';
+
+  // OneTrust stores consent in OnetrustActiveGroups.
+  oneTrustConsent = window.OnetrustActiveGroups.split(',');
+
+  const destinationPreferences = setSegmentDestinationPref(oneTrustConsent, SEGMENT_DESTINATIONS);
+
+  conditionallyLoadAnalytics({
+    writeKey: SEGMENT_WRITE_KEY,
+    destinations: SEGMENT_DESTINATIONS,
+    destinationPreferences
+  });
+}
+
+function setSegmentDestinationPref(oneTrustConsent, destinations) {
+  // maps Segment destination's category to OneTrust cookie category.
+  const segmentToOneTrust= {
+    'SMS & Push Notifications': 'C0001', // OneTrust Strictly Necessary
+    'Analytics': 'C0002', // OneTrust Performance
+    'Advertising': 'C0004' // OneTrust Targeting
+  }
+
+  const destinationPreferences = destinations
+    .map(function(dest) {
+      if ((dest.category in segmentToOneTrust) && oneTrustConsent.includes(segmentToOneTrust[dest.category])) {
+        return { [dest.id]: true };
+      } else {
+        // if we don't have a mapping for the Segment category, or we do but there's no OneTrust consent.
+        return { [dest.id]: false };
+      }
+    })
+    .reduce(
+      (acc, val) => {
+        return {
+          ...val,
+          ...acc
+        };
+      },
+      { "Segment.io": oneTrustConsent.some(d => d === SEGMENT_MAPPING) }
+    );
+
+  return destinationPreferences;
+}
+
+function conditionallyLoadAnalytics({
+  writeKey,
+  destinations,
+  destinationPreferences,
+}) {
+  let isAnythingEnabled = false;
+
+  for (const destination in destinationPreferences) {
+    const isEnabled = destinationPreferences[destination];
+    if (isEnabled) {
+      isAnythingEnabled = true;
+    }
+  }
+
+  if (isAnythingEnabled) {
+    window.analytics.load(writeKey, { integrations: destinationPreferences });
+  }
+
+  // if everything is disabled, don't load segment.
+}
+
+async function fetchDestinationForWriteKey(writeKey) {
+  const res = await window.fetch(
+    `https://cdn.segment.com/v1/projects/${writeKey}/integrations`
+  );
+
+  if (!res.ok) {
+    throw new Error(
+      `Failed to fetch Segment destinations: HTTP ${
+        res.status
+      } ${res.statusText}`
+    );
+  }
+
+  const destinations = await res.json();
+
+  // Rename creationName to id to abstract the weird data model
+  for (const destination of destinations) {
+    destination.id = destination.creationName;
+    delete destination.creationName;
+  }
+
+  return destinations;
+}
+
+}}();
+</script>
+
+<%% } %>
```

Comment:
We can likely move all of this script code into a proper JS file and initialize it on load (within the app)